dspace-server-webapp/src/test/java/org/dspace/app/rest/SitemapRestControllerIT.java

```diff
@@ -0,0 +1,167 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import javax.servlet.ServletException;
+
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.Item;
+import org.dspace.services.ConfigurationService;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.web.servlet.MvcResult;
+
+/**
+ * Integration test to test the /sitemaps/{name} endpoint, see {@link SitemapRestController}
+ *
+ * @author Maria Verdonck (Atmire) on 08/07/2020
+ */
+public class SitemapRestControllerIT extends AbstractControllerIntegrationTest {
+
+    @Autowired
+    ConfigurationService configurationService;
+
+    private final static String SITEMAPS_ENDPOINT = "sitemaps";
+
+    private Item item1;
+    private Item item2;
+
+    @Before
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+
+        configurationService.setProperty("sitemap.path", SITEMAPS_ENDPOINT);
+
+        context.turnOffAuthorisationSystem();
+
+        Community community = createCommunity(context).build();
+        Collection collection = createCollection(context, community).build();
+        this.item1 = createItem(context, collection)
+            .withTitle("Test 1")
+            .withIssueDate("2010-10-17")
+            .build();
+        this.item2 = createItem(context, collection)
+            .withTitle("Test 2")
+            .withIssueDate("2015-8-3")
+            .build();
+
+        runDSpaceScript("generate-sitemaps");
+
+        context.restoreAuthSystemState();
+    }
+
+    @After
+    public void destroy() throws Exception {
+        // delete sitemaps generated by tests in before
+        runDSpaceScript("generate-sitemaps", "-d");
+
+        super.destroy();
+    }
+
+    @Test
+    public void testSitemap_notValidSiteMapFile() throws Exception {
+        //** WHEN **
+        //We attempt to retrieve a non valid sitemap file
+        getClient().perform(get("/" + SITEMAPS_ENDPOINT + "/notValidSiteMapFile"))
+                   //** THEN **
+                   .andExpect(status().isNotFound());
+    }
+
+    @Test(expected = ServletException.class)
+    public void testSitemap_fileSystemTraversal_dspaceCfg() throws Exception {
+        //** WHEN **
+        //We attempt to use endpoint for malicious file system traversal
+        getClient().perform(get("/" + SITEMAPS_ENDPOINT + "/%2e%2e/config/dspace.cfg"));
+    }
+
+    @Test(expected = ServletException.class)
+    public void testSitemap_fileSystemTraversal_dspaceCfg2() throws Exception {
+        //** WHEN **
+        //We attempt to use endpoint for malicious file system traversal
+        getClient().perform(get("/" + SITEMAPS_ENDPOINT + "/%2e%2e%2fconfig%2fdspace.cfg"));
+    }
```

Comment:
I agree, this seems like it should be throwing a `ResourceNotFoundException` (or maybe responding with a simple 404 response code, i.e. `.andExpect(status().isNotFound());` ) based on the code in SitemapRestController here: https://github.com/DSpace/DSpace/pull/2858/files#diff-be5a61ded7d68e1169e3c3576ca9ce30R93

If that's not possible (for some reason), we could leave this as-is & add either a TODO or bug ticket. As-is, this test does prove that you cannot access other files from this endpoint, but the `@Test(expected=ServletException.class)` just seems like the wrong response.

In debugging this, we could add an additional simplistic test here to ensure an invalid filename (something like "notasitemap.html") also throws a `ResourceNotFoundException` or 404 response.

Reply:
This is valid behavior because it's an attempt to bypass security. The message returned is also: `The request was rejected because the URL contained a potentially malicious String "%2e"`
For non-attacks, just incorrect filenames, 404 is returned instead

Reply:
I'm OK with this for now. If we later decide to improve on it, we can do so. The bigger point is that this sort of request should throw an error & it obviously does. Thanks @benbosman